apps/rush/src/utilities/ShrinkwrapFile.ts

```diff
@@ -2,112 +2,41 @@
 // See LICENSE in the project root for license information.
 
 import * as colors from 'colors';
-import * as fsx from 'fs-extra';
-import * as os from 'os';
 import * as semver from 'semver';
 import npmPackageArg = require('npm-package-arg');
 
 import { Utilities, RushConstants } from '@microsoft/rush-lib';
 
-interface IShrinkwrapDependencyJson {
-  version: string;
-  from: string;
-  resolved: string;
-  dependencies: { [dependency: string]: IShrinkwrapDependencyJson };
-}
-
-interface IShrinkwrapJson {
-  name: string;
-  version: string;
-  dependencies: { [dependency: string]: IShrinkwrapDependencyJson };
-}
-
 /**
- * This class is a parser for NPM's npm-shrinkwrap.json file format.
+ * This class is a parser for both NPM's npm-shrinkwrap.json and PNPM's shrinkwrap.yaml file formats.
  */
-export default class ShrinkwrapFile {
-  private _shrinkwrapJson: IShrinkwrapJson;
-  private _alreadyWarnedSpecs: Set<string> = new Set<string>();
+export default abstract class ShrinkwrapFile {
+  protected _alreadyWarnedSpecs: Set<string> = new Set<string>();
 
-  public static loadFromFile(shrinkwrapJsonFilename: string): ShrinkwrapFile | undefined {
-    let data: string | undefined = undefined;
+  public static loadFromFile(shrinkwrapFilename: string): ShrinkwrapFile | undefined {
     try {
-      if (!fsx.existsSync(shrinkwrapJsonFilename)) {
-        return undefined; // file does not exist
-      }
-
-      // We don't use JsonFile/jju here because shrinkwrap.json is a special NPM file format
-      // and typically very large, so we want to load it the same way that NPM does.
-      data = fsx.readFileSync(shrinkwrapJsonFilename).toString();
-      if (data.charCodeAt(0) === 0xFEFF) {  // strip BOM
-        data = data.slice(1);
-      }
-
-      return new ShrinkwrapFile(JSON.parse(data));
-    } catch (error) {
-      throw new Error(`Error reading "${shrinkwrapJsonFilename}":` + os.EOL + `  ${error.message}`);
-    }
+      return require('./npm/NpmShrinkwrapFile').default.loadFromFile(shrinkwrapFilename);
+    } catch (e) { /* no-op */ }
+    try {
+      return require('./pnpm/PnpmShrinkwrapFile').default.loadFromFile(shrinkwrapFilename);
+    } catch (e) { /* no-op */ }
```

Comment:
This is sorta weird logic. Shouldn't it try only the tool that's configured in `rush.json`?